### ISSUE TITLE
variable "strategy" is not defined in CandlestickFinder.js

### ISSUE THREAD

**Issue by chiu0602** [anandanand84/technicalindicators#15]
Here is the stack trace:
```
ReferenceError: strategy is not defined
    at AbandonedBaby._getLastDataForCandleStick (...\node_modules\technicalindicators\lib\candlestick\CandlestickFinder.js:51:75)
    at AbandonedBaby.hasPattern (...\node_modules\technicalindicators\lib\candlestick\CandlestickFinder.js:45:43)
...
...
```

**Comment by anandanand84** [comment]
Hmmm, Do not use candlestick patterns yet, it is not complete, I'll add example in runkit once it is done, This is very basic there will be a lot of changes when we add more candlestick so the api may break.

**Comment by anandanand84** [comment]
I think you are using the old version, has pattern should work in latest version, all tests are passing. It should fail only in getAllPatternIndex.

**Comment by chiu0602** [comment]
I don't sure whether old version is used, but variable `strategy` cannot be found.

Even the pattern check haven't been fully implemented, I really can't wait for this interesting feature. Please let me put some comments here:
If array cannot fulfill `requiredCount`, exception should not be thrown. It is because logically it should be grouped into "pattern not found".
It is no doubt that your approach is right, just because I have written code as follow as a trial run::
```
var input = {
  date: [..., ..., ..., ...],
  open: [..., ..., ..., ...],
  close: [..., ..., ..., ...],
  high: [..., ..., ..., ...],
  low: [..., ..., ..., ...]
}
for(i =0 ... length) {
  let candlePattern;
  if(TI.AbandonedBaby.hasPattern(input)) candlePattern = ", AbandonedBaby";
  else if(TI.BearishEngulfingPattern.hasPattern(input)) candlePattern = ", BearishEngulfingPattern";
  else if(TI.BullishEngulfingPattern.hasPattern(input)) candlePattern = ", BullishEngulfingPattern";
  else if(TI.DarkCloudCover.hasPattern(input)) candlePattern = ", DarkCloudCover";
  else if(TI.Doji.hasPattern(input)) candlePattern = ", Doji";
  else if(TI.DownsideTasukiGap.hasPattern(input)) candlePattern = ", DownsideTasukiGap";
  else if(TI.DragonFlyDoji.hasPattern(input)) candlePattern = ", DragonFlyDoji";
  ...
  //output when pattern is found
  ...
  input.date.shift();
  input.open.shift();
  input.close.shift();
  input.high.shift();
  input.low.shift();
}
```
As candle patterns have no fixed `requiredCount`, it seems that the simple way to achieve what I want is to catch the exceptions.

Anyway, that is just my thought. Thanks for your great code.

**Comment by anandanand84** [comment]
This is fixed on 0.2.9. 

[sample](https://runkit.com/aarthiaradhana/bearishengulfingpattern)

**Comment by chiu0602** [comment]
Sorry may I know where is `strategy` defined? I saw it be used in `CandlestickFinder.js` but I am not able to find where it is declared. Thanks

**Comment by anandanand84** [comment]
CandlestickFinder is the parent class for all candlestick patterns, every candlestick pattern has a method logic, which is the strategyFn that is used. You can see this 

https://github.com/anandanand84/technicalindicators/blob/master/lib/candlestick/CandlestickFinder.js#L32

and 

https://github.com/anandanand84/technicalindicators/blob/master/lib/candlestick/BearishEngulfingPattern.js#L8

**Comment by chiu0602** [comment]
Please try to change the input as follows in the [sample](https://runkit.com/aarthiaradhana/bearishengulfingpattern) to:
```
var twoDayBearishInput = {
  open: [21.44],
  high: [25.10],
  close: [23.25],
  low: [20.82]
}
```
Then you will get what I mean.
Thanks!

**Comment by anandanand84** [comment]
Good catch!! It was due to dangerous copy paste. I have fixed it and added a test for it. Thanks. It should be available in version 0.2.10
